ex9/lib/main.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+void main() => runApp(MaterialApp(home: FirestoreFirstDemo(),));
+
+FirestoreFirstDemoState pageState;
+
+class FirestoreFirstDemo extends StatefulWidget {
+  @override
+  FirestoreFirstDemoState createState() {
+    pageState = FirestoreFirstDemoState();
+    return pageState;
+  }
+}
+
+class FirestoreFirstDemoState extends State<FirestoreFirstDemo> {
+  final firestoreInstance =  Firestore .instance;
+  TextEditingController textEditingController = TextEditingController();
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text("Firestore Chat!")),
+      body: SingleChildScrollView(
+        physics: ScrollPhysics(),
+        scrollDirection: Axis.vertical,
+        child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: <Widget>[
+              Container(
+                height: 450,
+                child: StreamBuilder<QuerySnapshot>(
+                  stream: Firestore.instance.collection("chat").snapshots(),
+                  builder: (BuildContext context,
+                      AsyncSnapshot<QuerySnapshot> snapshot) {
+                    if (snapshot.hasError) return Text("Error: ${snapshot.error}");
+                    switch (snapshot.connectionState) {
+                      case ConnectionState.waiting:
+                        return CircularProgressIndicator();
+                      default:
+                        return ListView(
+                          children: snapshot.data.documents
+                              .map((DocumentSnapshot document) {
+                            Timestamp tt = document["datetime"];
+                            DateTime dt = DateTime.fromMicrosecondsSinceEpoch(
+                                tt.microsecondsSinceEpoch);
+                            return Card(
```

Comment:
Can you move this into a new Widget called `MessageWidget`? VSC can do it for you.

<img width="247" alt="Screen Shot 2020-08-02 at 2 59 21 PM" src="https://user-images.githubusercontent.com/3399854/89116589-cb75c300-d4d0-11ea-9dd9-42ce683d53a7.png">